plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -759,4 +762,47 @@ public FunctionDependencyDeclaration getFunctionDependencies(ConnectorSession se
     {
         return FunctionDependencyDeclaration.NO_DEPENDENCIES;
     }
+
+    @Override
+    public synchronized TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        FakerTableHandle fakerTableHandle = (FakerTableHandle) tableHandle;
+        TableInfo info = tables.get(fakerTableHandle.schemaTableName());
+
+        if (!info.properties().containsKey(TableInfo.DEFAULT_LIMIT_PROPERTY)) {
+            return TableStatistics.empty();
```

Comment:
This is just a safeguard, right? If the limit is not set explicitly as a table property, it's inherited from the schema property, or the catalog config, so there's always some value. Maybe leave a code comment.

Reply:
hmm - I was getting empty value here unless I set it explicitly for the table.